crates/worktree/src/worktree.rs

```diff
@@ -3224,11 +3224,74 @@ async fn is_git_dir(path: &Path, fs: &dyn Fs) -> bool {
 async fn build_gitignore(abs_path: &Path, fs: &dyn Fs) -> Result<Gitignore> {
     let contents = fs.load(abs_path).await?;
     let parent = abs_path.parent().unwrap_or_else(|| Path::new("/"));
+
     let mut builder = GitignoreBuilder::new(parent);
     for line in contents.lines() {
         builder.add_line(Some(abs_path.into()), line)?;
     }
-    Ok(builder.build()?)
+
+    builder.build()?;
+
+    let mut combined_builder = GitignoreBuilder::new(parent);
+
+    let home = home_dir();
+    let repo_root = abs_path.parent().unwrap_or_else(|| Path::new("/"));
+    let repo_git_path = repo_root.join(".git");
+    let repo_git_config_path = repo_git_path.join("config");
+    
+    let mut global_ignore_path = None;
+    if let Ok(config_output) = std::process::Command::new("git")
+        .args(["config", "--file", &repo_git_config_path.to_string_lossy(), "core.excludesFile"])
+        .output()
```

Comment:
I think we should have an async `global_gitignore_path` method on the `Fs` trait, so that we don't actually run a command in tests. That'll also make it possible to cover global gitignore handling in the test you added.